src/renderer/src/components/MenuLink.tsx

```diff
@@ -0,0 +1,88 @@
+import { cloneElement, Fragment, type JSX } from 'react'
+
+import { motion } from 'motion/react'
+import { NavLink, type NavLinkProps, useMatch } from 'react-router'
+
+import { StyleHelper } from '@renderer/helpers/StyleHelper'
+
+import MdChevronRight from '@renderer/assets/images/md-chevron-right.svg?react'
+
+export type TMenuLinkProps = {
+  iconElement?: JSX.Element
+  rightElement?: JSX.Element | null
+  layoutId: string
+  disabled?: boolean
+  matchRoot?: string
+} & NavLinkProps
+
+export const MenuLink = ({
+  layoutId,
+  className,
+  iconElement,
+  disabled = false,
+  rightElement,
+  children,
+  to,
+  ...props
+}: TMenuLinkProps): JSX.Element => {
+  const rightElementWithDefault =
+    rightElement === undefined ? <MdChevronRight aria-hidden className="size-6" /> : rightElement
+
+  const match = useMatch(`${to}/*`)
+
+  const handleClick = (e: React.MouseEvent<HTMLAnchorElement, MouseEvent>) => {
+    if (disabled || match) {
+      e.preventDefault()
+    }
+  }
```

Comment:
Instead it, what do you think to use `pointer-events-none` and `tabIndex={-1}`? Because a link can be accessed with keyboard.